File: S5/Design and Analysis of Algorithms/LAB/lab 1/11.py
l=[]
for i in range(int(input('Enter the number of elements: '))):
    x=int(input("x"+str(i+1)+": "))
    y=int(input("y"+str(i+1)+": "))
    l.append((x,y))
x1,y1=l[0]
x2,y2=l[1]
slope=(y2-y1)/(x2-x1)
for i in range(2,len(l)):
    x1,y1=l[i-1]
    x2,y2=l[i]
    if slope!=(y2-y1)/(x2-x1):
        print('False')
        break
else:
    print('True')

File: S2/Computer Programming/LAB/Lab 6/11.c
//Write a program to sort an array of elements using Bubble sort.
#include<stdio.h>
void swap(int *a, int *b){
    int temp = *a;
    *a = *b;
    *b = temp;
}
void bubblesort(int arr[], int n){
    for (int i = 0; i < n - 1; i++){
        for (int j = 0; j < n - i - 1; j++){
            if (arr[j] > arr[j + 1]){
                swap(&arr[j], &arr[j + 1]);
            }
        }
    }
}
void printArray(int arr[], int n){
    for (int i = 0; i < n; i++){
        printf("%d ", arr[i]);
    }
}
int main(){
    int n;
    printf("Enter the size of the array: ");
    scanf("%d", &n);
    int arr[n];
    for (int i = 0; i < n; i++){
        printf("Enter number %d: ", i + 1);
        scanf("%d", &arr[i]);
    }
    printf("The array is: ");
    printArray(arr, n);
    printf("\n");
    bubblesort(arr, n);
    printf("The sorted array is: ");
    printArray(arr, n);
    return 0;
}